### PR TITLE
fix(layout): move gcode controls to dropdown; fix vertical layout

### LINKED ISSUE #690: Move the g-code preview view options to drop down settings button like other panels
### Is your feature request related to a problem? Please describe

image

I keep Fluidd open on my second monitor, which is in a vertical/portrait mode. This causes some issues with how Fluidd is rendered, like an extremely small g-code preview window.

### Describe the solution you'd like

image

If the view options were moved to a drop down settings button (like the console panel), there would be a lot more space for a larger g-code preview window.

### Describe alternatives you've considered

None.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matmen**
Moves the gcode preview related settings to a dropdown and fixes the layout to be compatible with vertical monitors
image
image

Not sure if the "Load current file" and "Clear file" buttons should also go there, LMK what you think..

Closes #690

### CODE REVIEW

**Comment by matmen** [comment]
I think the "Load current file" button would fit quite well into the header, like this;
image

**Comment by pedrolamas** [comment]
> I think the "Load current file" button would fit quite well into the header

I was about to say that I was not very inclined to have those buttons on the dropdown, but this alternative looks really good to me!

I don't have a problem with the preview being that big, but I do get your point! Maybe we make it 16:9 but with a max height?

**Comment by matmen** [comment]
> I don't have a problem with the preview being that big, but I do get your point! Maybe we make it 16:9 but with a max height?

Sounds good. With a max height of 50% viewport height this would look like this:

Landscape orientation: Whole control still in view, preview not too small
image

Portrait orientation: Everything in view, preview filling whole empty space
image

**Comment by pedrolamas** [comment]
@matmen Removing myself from review as this is still on "draft" state.

**Review by pedrolamas — CHANGES_REQUESTED** [review]
We might want to change the layout for mobile...

image

Maybe show the layer counters + "clear" button above the controls, or vice versa?

File: src/components/widgets/gcode-preview/GcodePreviewCard.vue [hunk 1]
```diff
@@ -5,6 +5,24 @@
...
   >
+    <template #menu>
+      <app-btn
+        :disabled="!printerFile || printerFileLoaded"
+        color="primary"
+        small
+        @click="loadCurrent"
+      >
+        {{ $t('app.gcode.btn.load_current_file') }}
+      </app-btn>
```
[the remark this answers is not in the document]
Comment (matmen) [reply src/components/widgets/gcode-preview/GcodePreviewCard.vue:16]: Oh yeah, I was using that before, but it collapsed the controls in vertical mode too. I'll add that back

**Review by pedrolamas — APPROVED** [review]

**Comment by alexrmay91** [comment]
I know this is already closed and I'm late to the party, but an option (probably best in the main fluid settings) could be "automatically load current file on print start" and "automatically follow progress on file load". If this should be a separate suggestion that can be evaluated for a future milestone, let me know and I'd be happy to type it out.

**Pull request merged**